### ISSUE TITLE
Add `error=false` URL parameter

### ISSUE THREAD

**Issue by jomo** [crafatar/crafatar#87]
Not sure about the name, but there are some sites (forums) that proxy images.

When we return steve/alex with 404, or the (potentially) outdated avatar with a 50x, these services will not proxy the image.

~~GitHub's usercontent proxy does the same:~~ (atmos/camo#77)

```Markdown
![img](https://crafatar.com/avatars/0)
```
![img](https://crafatar.com/avatars/0)

**Comment by Jake0oo0** [comment]
Ugh..it's hard to decide what the right thing to do here is. Most people will immediately be confused as to why steves are showing, and think it's broken. Just setting a 200 http code could break apis that check for the status code. Maybe make the parameter show the actual code, rather than the opposite.

**Comment by jomo** [comment]
I'm not sure what you suggest.. So we should always return 200 unless that extra param is set?

**Comment by jomo** [comment]
I see two options here:

0. Send _all_ images with 200 HTTP code
0. Send images with proper HTTP code (200, steve = 404, mojang issues = 503)

We need a URL parameter to toggle between those two. The first one ~~seems~~ is wrong, so using it as the default would be bad.

**Comment by jomo** [comment]
We're now [using 200 for fallback images](111), so we're only left with the 503 problem.

Apparently Chrome(ium) doesn't render images with 4xx or 5xx status code.

Seems like chromium intends to render images regardless of the status code, so the bug has been confirmed:
> That's surprising. We explicitly try to ensure that we show images regardless of status code error, but I guess that logic must have regressed somehow.

More details: [chromium-464515](https://code.google.com/p/chromium/issues/detail?id=464515). We're not alone:
> This is a big problem for us as we have a user ID photo system that is designed to 404 and show a default image. Pages full of broken images aren't really acceptable. I'm now going to have to break best practices and make the 404 image return a 200 until this is fixed


**tl;dr** Browsers should do fine with the 503, I think having this parameter optional should work for image caching proxies (i.e. GitHub, etc).

How about calling it `error=false`?


----

Actually, I'm unsure about this whole thing. Image cache/proxies should forward 404 if the `Content-Type` is `image/whatever` – Other User-Agents (i.e. Browsers) do that as well.

I'll probably open an issue at [atmos/camo](/atmos/camo). What do you think @Jake0oo0?

**Comment by jomo** [comment]
here it is: atmos/camo#75

**Comment by jomo** [comment]
Closing because it's not a bug on our side. People need to fix their image proxies.
